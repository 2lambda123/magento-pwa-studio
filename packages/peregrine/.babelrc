{
  "plugins": [
    "transform-object-rest-spread",
    "transform-class-properties",
    ["transform-react-jsx", { "pragma": "createElement" }],
    ["transform-runtime", {
      "helpers": true,
      "polyfill": false, // polyfills will be handled by preset-env
      "regenerator": false
    }]
  ],
  "presets": [
    ["env", {
      "targets": {
        "browsers": ["> 5%"]
      },
      "modules": false
    }]
  ],
  "env": {
    "test": {
      "plugins": [
        "transform-es2015-modules-commonjs",
        "transform-object-rest-spread",
        "transform-class-properties",
        ["transform-react-jsx", { "pragma": "createElement" }]
      ]
    }
  }
}
