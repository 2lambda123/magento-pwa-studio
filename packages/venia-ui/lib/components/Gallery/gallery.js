import React from 'react';
import { string, shape, array, number } from 'prop-types';

import { mergeClasses } from '../../classify';
import GalleryItems, { emptyData } from './items';
import defaultClasses from './gallery.css';

const Gallery = props => {
    const classes = mergeClasses(defaultClasses, props.classes);

    const { data, pageSize } = props;
    const hasData = Array.isArray(data) && data.length;
    const items = hasData ? data : emptyData;

    return (
        <div className={classes.root}>
            <div className={classes.items}>
                <GalleryItems items={items} pageSize={pageSize} />
            </div>
        </div>
    );
};

Gallery.propTypes = {
    classes: shape({
        filters: string,
        items: string,
        pagination: string,
        root: string
    }),
    data: array,
    pageSize: number
};

Gallery.defaultProps = {
    data: emptyData
};

export default Gallery;
