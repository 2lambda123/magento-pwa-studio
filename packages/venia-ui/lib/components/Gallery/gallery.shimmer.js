import React from 'react';
import { shape, string, array, object } from 'prop-types';
import { mergeClasses } from '../../classify';
import ItemShimmer from './item.shimmer';
import defaultClasses from './gallery.css';

const GalleryShimmer = (props) => {
    const { items, itemProps } = props
    const classes = mergeClasses(defaultClasses, props.classes);

    return (
        <div className={classes.root} aria-live="polite" aria-busy="true">
            <div className={classes.items}>
                {items.map((item, index) => {
                    return <ItemShimmer key={`item_${index}`} {...itemProps} />;
                })}
            </div>
        </div>
    );
};

GalleryShimmer.defaultProps = {
    classes: {},
    items: [],
    itemProps: {}
};

GalleryShimmer.propTypes = {
    classes: shape({
        root: string,
        items: string
    }),
    items: array,
    itemProps: shape({
        classes: object
    })
};

export default GalleryShimmer;
