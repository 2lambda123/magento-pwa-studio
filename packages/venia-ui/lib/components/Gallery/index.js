export { default } from './gallery';
export { default as GalleryShimmer } from './gallery.shimmer';
export { default as ItemShimmer } from './item.shimmer';
