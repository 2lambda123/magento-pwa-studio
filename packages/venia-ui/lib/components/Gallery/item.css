.root {
    align-content: start;
    display: grid;
    row-gap: 0.5rem;
}

.images {
    display: grid;
    grid-template-areas: 'main';
}

.imageContainer {
    grid-area: main;
}

.image {
    display: block;
    height: 100%;
    object-fit: contain;
    transition: opacity 512ms ease-out;
    width: 100%;
}

.imageLoaded {
    composes: loaded from '../Image/image.css';
    opacity: 1;
}

.imageNotLoaded {
    composes: notLoaded from '../Image/image.css';
    opacity: 0;
}

.imagePlaceholder {
    composes: image;
    background-color: rgb(var(--venia-global-color-gray));
}

.actionsContainer {
    column-gap: 0.5rem;
    display: flex;
    justify-content: space-between;
}

.actionsContainer:empty {
    display: none;
}

.name {
    font-weight: var(--venia-global-fontWeight-semibold);
}

.name,
.price {
    font-size: var(--venia-typography-detail-XL-fontSize);
    min-height: 1rem;
}
