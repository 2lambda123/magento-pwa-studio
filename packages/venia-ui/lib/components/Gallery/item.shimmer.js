import React from 'react';
import { shape, string } from 'prop-types';
import { transparentPlaceholder } from '@magento/peregrine/lib/util/images';
import { useStyle } from '../../classify';

import Shimmer from '../Shimmer';
import Image from '../Image';
import defaultClasses from './item.css';

const GalleryItemShimmer = props => {
    const classes = useStyle(defaultClasses, props.classes);

    return (
        <div className={classes.root} aria-live="polite" aria-busy="true">
            <Shimmer>
                <Image
                    alt="Placeholder for gallery item image"
                    classes={{
                        image: classes.image,
                        root: classes.imageContainer
                    }}
                    src={transparentPlaceholder}
                />
            </Shimmer>
            <Shimmer width="100%" />
            <Shimmer width={3} />
        </div>
    );
};

GalleryItemShimmer.propTypes = {
    classes: shape({
        root: string
    })
};

export default GalleryItemShimmer;
