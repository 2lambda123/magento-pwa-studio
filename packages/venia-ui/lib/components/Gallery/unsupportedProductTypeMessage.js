import React from 'react';
import { Info } from 'react-feather';

import { FormattedMessage } from 'react-intl';

const InfoIcon = <Icon size={20} src={Info} />;


const UnsupportedProductTypeMessage = props => {
    const talonProps = useUnsupportedProductTypeMessage(props);
    const {
        isUnsupportedProductType, 
        
        
    } = talonProps;
    const classes = useStyle(defaultClasses, props.classes);
    
    const buttonClass = isSelected ? classes.root_selected : classes.root;

    return (
        <button className={buttonClass} {...buttonProps}>
            {props.icon} {buttonText}
        </button>
    );
};

export default AddToListButton;

AddToListButton.defaultProps = {
    icon: HeartIcon
};

AddToListButton.propTypes = {
    afterAdd: func,
    beforeAdd: func,
    classes: shape({
        root: string,
        root_selected: string
    }),
    icon: element
};
