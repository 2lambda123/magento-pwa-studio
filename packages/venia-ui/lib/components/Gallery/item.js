import React from 'react';
import { string, number, shape } from 'prop-types';
import { Link } from 'react-router-dom';
import Price from '@magento/venia-ui/lib/components/Price';
import { UNCONSTRAINED_SIZE_KEY } from '@magento/peregrine/lib/talons/Image/useImage';
import { useGalleryItem } from '@magento/peregrine/lib/talons/Gallery/useGalleryItem';
import resourceUrl from '@magento/peregrine/lib/util/makeUrl';

import { useStyle } from '../../classify';
import Image from '../Image';
import GalleryItemShimmer from './item.shimmer';
import defaultClasses from './item.css';
import WishlistGalleryButton from '../Wishlist/AddToListButton';

// The placeholder image is 4:5, so we should make sure to size our product
// images appropriately.
const IMAGE_WIDTH = 300;
const IMAGE_HEIGHT = 375;

// Gallery switches from two columns to three at 640px.
const IMAGE_WIDTHS = new Map()
    .set(640, IMAGE_WIDTH)
    .set(UNCONSTRAINED_SIZE_KEY, 840);

const GalleryItem = props => {
    const { handleLinkClick, item, wishlistButtonProps } = useGalleryItem(
        props
    );

    const classes = useStyle(defaultClasses, props.classes);

    if (!item) {
        return <GalleryItemShimmer classes={classes} />;
    }

    const { name, price, small_image, url_key, url_suffix } = item;
    const { url: smallImageURL } = small_image;
    const productLink = resourceUrl(`/${url_key}${url_suffix || ''}`);

    const wishlistButton = wishlistButtonProps ? (
        <WishlistGalleryButton {...wishlistButtonProps} />
    ) : null;

    return (
        <div className={classes.root} aria-live="polite" aria-busy="false">
            <Link
                onClick={handleLinkClick}
                to={productLink}
                className={classes.images}
            >
                <Image
                    alt={name}
                    classes={{
                        image: classes.image,
                        loaded: classes.imageLoaded,
                        notLoaded: classes.imageNotLoaded,
                        root: classes.imageContainer
                    }}
                    height={IMAGE_HEIGHT}
                    resource={smallImageURL}
                    widths={IMAGE_WIDTHS}
                />
            </Link>
            <Link
                onClick={handleLinkClick}
                to={productLink}
                className={classes.name}
            >
                <span>{name}</span>
            </Link>
            <div className={classes.price}>
                <Price
                    value={price.regularPrice.amount.value}
                    currencyCode={price.regularPrice.amount.currency}
                />
            </div>
            <div className={classes.actionsContainer}>{wishlistButton}</div>
        </div>
    );
};

GalleryItem.propTypes = {
    classes: shape({
        image: string,
        imageLoaded: string,
        imageNotLoaded: string,
        imageContainer: string,
        images: string,
        name: string,
        price: string,
        root: string
    }),
    item: shape({
        id: number.isRequired,
        name: string.isRequired,
        small_image: shape({
            url: string.isRequired
        }),
        url_key: string.isRequired,
        price: shape({
            regularPrice: shape({
                amount: shape({
                    value: number.isRequired,
                    currency: string.isRequired
                }).isRequired
            }).isRequired
        }).isRequired
    }),
    storeConfig: shape({
        magento_wishlist_general_is_enabled: string.isRequired
    })
};

export default GalleryItem;
