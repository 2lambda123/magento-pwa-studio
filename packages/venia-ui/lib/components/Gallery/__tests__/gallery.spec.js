import React from 'react';
import { createTestInstance } from '@magento/peregrine';

import Gallery from '../gallery';
jest.mock('../items', () => 'GalleryItems');

const classes = { root: 'foo' };
const items = [
    {
        id: 1,
        name: 'Test Product 1',
        small_image: {
            url: '/test/product/1.png'
        },
        price: {
            regularPrice: {
                amount: {
                    value: 100,
                    currency: 'USD'
                }
            }
        }
    },
    {
        id: 2,
        name: 'Test Product 2',
        // Magento 2.3.0 schema for testing backwards compatibility
        small_image: '/test/product/2.png',
        price: {
            regularPrice: {
                amount: {
                    value: 100,
                    currency: 'USD'
                }
            }
        }
    }
];

test('renders if `data` is an empty array', () => {
    const wrapper = createTestInstance(<Gallery classes={classes} data={[]} />);
    expect(wrapper.toJSON()).toMatchSnapshot();
});

test('renders if `data` is an array of objects', () => {
    const wrapper = createTestInstance(
        <Gallery classes={classes} data={items} />
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
});
