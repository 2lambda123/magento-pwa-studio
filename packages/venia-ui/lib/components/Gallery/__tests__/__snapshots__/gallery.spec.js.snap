// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders if \`data\` is an array of objects 1`] = `
<div
  className="foo"
>
  <div>
    <GalleryItems
      items={
        Array [
          Object {
            "id": 1,
            "name": "Test Product 1",
            "price": Object {
              "regularPrice": Object {
                "amount": Object {
                  "currency": "USD",
                  "value": 100,
                },
              },
            },
            "small_image": Object {
              "url": "/test/product/1.png",
            },
          },
          Object {
            "id": 2,
            "name": "Test Product 2",
            "price": Object {
              "regularPrice": Object {
                "amount": Object {
                  "currency": "USD",
                  "value": 100,
                },
              },
            },
            "small_image": "/test/product/2.png",
          },
        ]
      }
    />
  </div>
</div>
`;

exports[`renders if \`data\` is an empty array 1`] = `
<div
  className="foo"
>
  <div>
    <GalleryItems />
  </div>
</div>
`;
